src/lib/utils/config/src/statics.rs

```diff
@@ -74,6 +74,94 @@ fn create_config() -> ServerConfig {
     }
 }
 
+fn create_whitelist() -> DashMap<u128, String> {
+    let whitelist_location = get_root_path().join("whitelist.txt");
+    if !whitelist_location.exists() {
+        write_whitelist_to_file();
+    }
+
+    let mut file = match File::open(&whitelist_location) {
+        Ok(file) => file,
+        Err(e) => {
+            error!("Could not open white-list file: {e}");
+            return DashMap::new();
+        }
+    };
+
+    let mut whitelist_str = String::new();
+    if let Err(e) = file.read_to_string(&mut whitelist_str) {
+        error!("Could not read white-list file: {e}");
+        return DashMap::new();
+    }
+
+    if whitelist_str.is_empty() {
+        return DashMap::new();
+    }
+
+    //read and split the file, its username:uuid format
+    let whitelist: DashMap<u128, String> = DashMap::new();
+    for (i, line) in whitelist_str.lines().enumerate() {
+        let line = line.trim();
+        if line.is_empty() || line.starts_with('#') {
+            continue;
+        }
+
+        let mut split = line.split(':');
```

Comment:
Requiring both a username and uuid can be a hassle. uuids can be matched with regex pretty easily so you can just go through each line, check if it matches a regex and if it does, its a uuid, otherwise interpret as a username. Then on login, just check if either the username or the uuid is in the whitelist, that way you can have either a player's uuid or their username and add them to the whitelist.